pkg/controller/controllerutil/controllerutil.go

```diff
@@ -195,16 +197,7 @@ const ( // They should complete the sentence "Deployment default/foo has been ..
 func CreateOrUpdate(ctx context.Context, c client.Client, obj client.Object, f MutateFn) (OperationResult, error) {
 	key := client.ObjectKeyFromObject(obj)
 	if err := c.Get(ctx, key, obj); err != nil {
-		if !apierrors.IsNotFound(err) {
-			return OperationResultNone, err
-		}
-		if err := mutate(f, key, obj); err != nil {
-			return OperationResultNone, err
-		}
-		if err := c.Create(ctx, obj); err != nil {
-			return OperationResultNone, err
-		}
-		return OperationResultCreated, nil
+		return CreateIfNotExist(ctx, c, obj, f)
```

Comment:
`CreateIfNotExist` and `CreateOrUpdate` are similar. I am wondering if we need a separate helper for `CreateIfNotExist`. One can use both the helpers to do the same, right?

Reply:
I agree. Is there a use case for `CreateIfNotExist` that is not already covered by `CreateOrUpdate`?

Reply:
@joelanford the one comment I got from the developers that wrote the Create*IfNotExist utilties was the CreateOrUpdate was hard to use with the mutations. So maybe there is a place to better document that call with better examples.